qa/os/src/test/java/org/elasticsearch/packaging/test/DockerTests.java

```diff
@@ -347,6 +414,158 @@ public void test92ElasticsearchNodeCliPackaging() {
+     * @see <a href="https://github.com/opencontainers/image-spec/blob/master/annotations.md">Opencontainers Annotations</a>
+     */
+    public void test110OrgOpencontainersLabels() throws Exception {
+        final Map<String, String> labels = getImageLabels(distribution);
+
+        final Map<String, String> staticLabels = new HashMap<>();
+        staticLabels.put("title", "Elasticsearch");
+        staticLabels.put("url", "https://www.elastic.co/products/elasticsearch");
+        staticLabels.put("documentation", "https://www.elastic.co/guide/en/elasticsearch/reference/index.html");
+        staticLabels.put("source", "https://github.com/elastic/elasticsearch");
+        staticLabels.put("vendor", "Elastic");
+
+        if (distribution.isOSS()) {
+            staticLabels.put("licenses", "Apache-2.0");
+        } else {
+            staticLabels.put("licenses", "Elastic-License");
+        }
+
+        // TODO: we should check the actual version value
+        final Set<String> dynamicLabels = Set.of("created", "revision", "version");
+
+        final String prefix = "org.opencontainers.image";
+
+        staticLabels.forEach((suffix, value) -> {
+            String key = prefix + "." + suffix;
+            assertThat(labels, hasKey(key));
+            assertThat(labels.get(key), equalTo(value));
+        });
+
+        dynamicLabels.forEach(label -> {
+            String key = prefix + "." + label;
+            assertThat(labels, hasKey(key));
+        });
+    }
+
+    /**
+     * Check that the container logs contain the expected content for Elasticsearch itself.
+     */
+    public void test120DockerLogsIncludeElasticsearchLogs() throws Exception {
+        waitForElasticsearch(installation);
+        final Result containerLogs = getContainerLogs();
+
+        assertThat("Container logs don't contain abbreviated class names", containerLogs.stdout, containsString("o.e.n.Node"));
+        assertThat("Container logs don't contain INFO level messages", containerLogs.stdout, containsString("INFO"));
+    }
+
+    /**
+     * Check that the Java process running inside the container has the expect PID, UID and username.
+     */
+    public void test130JavaHasCorrectPidAndOwnership() {
+        final List<String> processes = sh.run("ps -o pid,uid,user -C java").stdout.lines().skip(1).collect(Collectors.toList());
+
+        assertThat("Expected a single java process", processes, hasSize(1));
+
+        final String[] fields = processes.get(0).trim().split("\\s+");
+
+        assertThat(fields, arrayWithSize(3));
+        assertThat("Incorrect PID", fields[0], equalTo("1"));
+        assertThat("Incorrect UID", fields[1], equalTo("1000"));
+        assertThat("Incorrect username", fields[2], equalTo("elasticsearch"));
+    }
+
+    /**
+     * Check the provided version of Java. This guards against accidentally shipping images with
+     * the wrong version.
+     */
+    public void test131JavaHasExpectedVersion() {
```

Comment:
Since java is bundled now in all our distributions, I'm not sure what this check really provides? I would hate for it to be yet another thing that must be bumped when we upgrade java.